examples/sudoku/server/sudoku.hpp

```diff
@@ -0,0 +1,216 @@
+//  Copyright (c) 2016 Satyaki Upadhyay
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_SUDOKU_EXAMPLE_SERVER)
+#define HPX_SUDOKU_EXAMPLE_SERVER
+
+#include <iostream>
+
+#include <hpx/include/async.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/hpx_fwd.hpp>
+#include <hpx/include/components.hpp>
+#include <hpx/runtime/applier/applier.hpp>
+#include <hpx/runtime/threads/thread_data.hpp>
+#include <hpx/runtime/components/component_type.hpp>
+#include <hpx/runtime/actions/component_action.hpp>
+#include <hpx/runtime/serialization/serialize.hpp>
+
+namespace sudoku
+{
+    typedef std::vector<std::size_t> list_type;
+
+namespace server
+{
+    class board
+        : public hpx::components::component_base<board>
+    {
+    private:
+        list_type list_;
```

Comment:
The 9X9 board is represented as a list of 81 elements. Position (i, j) on the board corresponds to (9*i+j) in the list (0-indexed).